src/main/java/io/supertokens/session/Session.java

```diff
@@ -229,17 +236,18 @@ public static SessionInformationHolder getSession(AppIdentifier appIdentifier, M
         AccessTokenInfo accessToken = AccessToken.getInfoFromAccessToken(appIdentifier, main,
                 token,
                 doAntiCsrfCheck && enableAntiCsrf);
-        TenantIdentifier tenantIdentifier = accessToken.tenantIdentifier;
+        TenantIdentifierWithStorage tenantIdentifierWithStorage = accessToken.tenantIdentifier.withStorage(
+                StorageLayer.getStorage(accessToken.tenantIdentifier, main));
 
```

Comment:
check if the input app identifier has the same appid as the one from the access token in the getInfoFromAccessToken function